### ISSUE TITLE
vcfR2DNAbin with haploid data returns NaN

### ISSUE THREAD

**Issue by knausb** [knausb/vcfR#108]
When using `vcfR2DNAbin()` on haploid data it appear to return a DNAbin object that is entirely `NaN`.

```
# Load test data
data(vcfR_test)
  
# Haploidize test data
my_non_gt <- extract.gt(vcfR_test, extract = FALSE)
my_gt <- extract.gt(vcfR_test)
my_gt <- unlist(lapply(strsplit(my_gt, split = "[/|]"), function(x){x[1]}))
# https://stackoverflow.com/a/35589023
my_non_gt <- paste(my_gt, my_non_gt, sep = ":")
dim(my_non_gt) <- dim(vcfR_test@gt[,-1])
vcfR_test@gt[,-1] <- my_non_gt

vcfR2DNAbin(vcfR_test)
```

```
After extracting indels, 4 variants remain.
3 DNA sequences in binary format stored in a matrix.

All sequences of same length: 4 

Labels:
NA00001
NA00002
NA00003

Base composition:
  a   c   g   t 
NaN NaN NaN NaN 
```

We should get the alleles.

**Comment by knausb** [comment]
The issue may be due to `extract.haps()` returning numerics with haploid data.

First demonstrate expected behaviour using diploid data.

```{r}
data(vcfR_test)
extract.haps(vcfR_test)
```

Then haploid data.

```
# Haploidize test data
my_non_gt <- extract.gt(vcfR_test, extract = FALSE)
my_gt <- extract.gt(vcfR_test)
my_gt <- unlist(lapply(strsplit(my_gt, split = "[/|]"), function(x){x[1]}))
# https://stackoverflow.com/a/35589023
my_non_gt <- paste(my_gt, my_non_gt, sep = ":")
dim(my_non_gt) <- dim(vcfR_test@gt[,-1])
vcfR_test@gt[,-1] <- my_non_gt

extract.haps(vcfR_test)
```

**Comment by knausb** [comment]
`extract.haps()` did not include the parameter `return.alleles = TRUE` in it's call to `extract.gt()` in the haploid branch of the function. This parameter has now been added. This also affects `vcfR2DNAbin()` which calls this function. Fixed at: 893b63e66d30fc1f25074c33cfea409ce33611f9.
